File: excersize6/typescript/6.ts
export function add(a: number, b: number): number {
    return a + b;
}

export function subtract(a: number, b: number): number {
    return a - b;
}

export function multiply(a: number, b: number): number {
    return a * b;
}

export function divide(a: number, b: number): number {
    return a / b;
}

export function mod(a: number, b: number): number {
    return a % b;
}

export function andFunc(a: boolean, b: boolean): boolean {
    return a && b;
}

export function orFunc(a: boolean, b: boolean): boolean {
    return a || b;
}

export function notFunc(a: boolean): boolean {
    return !a;
}

export function greaterThan(a: number, b: number): boolean {
    return a > b;
}

export function lessEqual(a: number, b: number): boolean {
    return a <= b;
}

export function equal(a: any, b: any): boolean {
    return a == b;
}

const a = 5, b = 3, c = 7, d = 4, e = 2, f = 10;
const x = true, y = false, z = true, isAdmin = false;
const tg =0;
const expected2 = (a - b) * (c + d) <= e && z && !x;
const actual2= andFunc(
    andFunc(
        lessEqual(
            multiply(
                subtract(a, b),
                add(c, d)
            ),
            e
        ),
        z
    ),
    notFunc(x)
);
console.log(expected2,actual2)
const expected3 = (a + b + c) % d == 0 || (f > e && isAdmin);
const actual3 = orFunc(
    equal(mod(add(add(a, b), c), d),0),
    andFunc(
        greaterThan(f, e),
        isAdmin
    )
);
console.log(expected3, actual3)
const actual4 = andFunc(
    notFunc(
        orFunc(x, y)
    ),
    greaterThan(
        multiply(a, a),
        multiply(b, c)
    )
);
console.log(actual4,actual4)

File: excersize6/python/6.py
from python.script import *
a, b, c, d, e, f = 5, 3, 7, 4, 2, 10
x, y, z, is_admin = True, False, True, False
expected1 = (a + b * c) > (d - e) and x or y
actual1 = or_func( and_func(  greater_than( add(a, multiply(b, c)),  subtract(d, e)),x),y)
print(f"1) Expected: {expected1}, Actual: {actual1}")

expected2 = (a - b) * (c + d) <= e and z and not x
actual2 =and_func(
    and_func(
        less_equal(
            multiply(
             subtract(a, b),
             add(c,d)),
                   e), z),
    not_func(x)
)
print(f" {expected2},{actual2}")

expected3 = (a + b + c) % d == 0 or (f > e and is_admin)
actual3=and_func(

    or_func(equal
            (mod
             (add
              
              (add(a,b),c))
              ,0),
            greater_than(f,e))
    ,is_admin)
print(f"{expected3},{actual3}")

expected4 = not (x or y) and (a * a > b * c)
actual4 = and_func(not_func(or_func(x,y)),(greater_than(multiply(a,a),multiply(b,c))))
print(f"{expected4},{actual4}")